oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/nodetype/write/NodeTypeRegistry.java

```diff
@@ -89,9 +93,34 @@ public static void register(Root root, InputStream input, String systemId) {
 
     private void registerNodeTypes(InputStream stream, String systemId) {
         try {
-            CndImporter.registerNodeTypes(
-                    new InputStreamReader(stream, Charsets.UTF_8),
-                    systemId, ntMgr, nsReg, vf, false);
+            Reader reader = new InputStreamReader(stream, Charsets.UTF_8);
+            // OAK-9134: nt:frozenNode is not implementing mix:referenceable from JCR 2.0.
+            // This system property allows to add it back when initializing a repository.
+            final String referenceableFrozenNodeProperty = "oak.referenceableFrozenNode";
+            final boolean referenceableFrozenNode;
+            if (System.getProperty(referenceableFrozenNodeProperty) == null) {
+                // the default for referenceableFrozenNode is true in the 1.22 branch.
+                // this is in contrast to it being false in newer versions.
+                // the reason for choosing true as the default is to maintain higher
+                // backwards compatibility and minimize an otherwise high impact in this branch.
+                referenceableFrozenNode = true;
+            } else {
+                referenceableFrozenNode = Boolean.getBoolean(referenceableFrozenNodeProperty);
+            }
```

Comment:
To reduce divergence, what do you think about backporting OAK-9010 and then use a `SystemPropertySupplier`? The only difference then would be the default value when calling the factory method.

Reply:
+1, good point on reducing divergence

Reply:
turns out SystemPropertySupplier is in oak-store-document so not available here .. unless we move that class somewhere more generic ..

Reply:
Created a PR suggesting such a move in https://issues.apache.org/jira/browse/OAK-9352